pyiron_base/project/data.py

```diff
@@ -0,0 +1,51 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+"""
+A modified data container for storing associating data with a project.
+
+Spec:
+
+The `Project` class should have data associated with it which can be stored to-file in our supported format(s) (right
+now just hdf5). This should meet the following requirements:
+
+- Data storage is immediately accessible, i.e. appearing in the project tab-completion module
+- Data stored there should be readable and writeable with a single parameter-free call
+- When instantiated, new projects should automatically read any available data
+- The `Project` tab completion menu should not get cluttered (i.e. nest everything under `Project.foo`)
+"""
+
+from pyiron_base.generic.inputlist import InputList
+from pyiron_base.generic.hdfio import ProjectHDFio
+
+__author__ = "Liam Huber"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Liam Huber"
+__email__ = "huber@mpie.de"
+__status__ = "production"
+__date__ = "Feb 19, 2021"
+
+
+class ProjectData(InputList):
+    def __new__(cls, *args, **kwargs):
+        instance = super().__new__(cls, *args, **kwargs)
+        object.__setattr__(instance, "project", None)
+        return instance
+
+    def __init__(self, project, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.project = project
+
+    def read(self):
+        hdf = ProjectHDFio(self.project, file_name="project_data")
```

Comment:
why not `self.project.create_hdf(self.project.path, 'project_data)`  which opens the `hdf` at `h5_path='/project_data'`?

Reply:
I have one small objection to this, and that is `Project.create_hdf` re-instantiates another `Project` inside itself, which feels wasteful when that would just be a duplicate of the project calling the method.

I'd still be open to this change, but unfortunately it raises the error caught by the if-clause that immediately follows `if self.table_name not in hdf.list_groups():`.  So does

```python
hdf = ProjectHDFio(self.project, file_name="project_data", h5_path="/project_data")
```

So somehow providing the `h5_path` breaks things.

Reply:
Ok, actually, I just realize I don't like my idea then. Since the `Project` is hard coded to be a `base` project and `self.__class__` is not available in a `@staticmethod`.